src/rules/__tests__/createEvaluateConditions.test.js

```diff
@@ -37,114 +37,99 @@ describe('createEvaluateConditions returns a function that when called', functio
       return result;
     };
 
-    var logConditionNotMet = emptyFn;
-    var logConditionError = emptyFn;
+    var logConditionNotMetSpy = jasmine.createSpy();
+    var logConditionErrorSpy = jasmine.createSpy();
+    var result = createEvaluateConditions(
+      executeDelegateModule,
+      isConditionMet,
+      logConditionNotMetSpy,
+      logConditionErrorSpy
+    )(rule);
 
-    expect(
-      createEvaluateConditions(
-        executeDelegateModule,
-        isConditionMet,
-        logConditionNotMet,
-        logConditionError
-      )(rule)
-    ).toBeTrue();
+    expect(result).toBeTrue();
+    expect(logConditionNotMetSpy).not.toHaveBeenCalled();
+    expect(logConditionErrorSpy).not.toHaveBeenCalled();
   });
 
-  it('returns false when conditions are not met', function () {
-    var executeDelegateModule = function (condition) {
-      return condition.modulePath === 'condition1';
-    };
-    var isConditionMet = function () {
-      return false;
-    };
-
-    var logConditionNotMet = emptyFn;
-    var logConditionError = emptyFn;
+  it('handles a condition returning a promise', function () {
+    var executeDelegateModule = function () {
+      // Not using a real promise to ensure that the logic detects
+      // objects that are promise-like.
+      return {
+        then: function () {}
+      };
+    };
+    var isConditionMet = jasmine.createSpy();
+    var logConditionNotMetSpy = jasmine.createSpy();
+    var logConditionErrorSpy = jasmine.createSpy();
+    var result = createEvaluateConditions(
+      executeDelegateModule,
+      isConditionMet,
+      logConditionNotMetSpy,
+      logConditionErrorSpy
+    )(rule);
 
-    expect(
-      createEvaluateConditions(
-        executeDelegateModule,
-        isConditionMet,
-        logConditionNotMet,
-        logConditionError
-      )(rule)
-    ).toBeFalse();
+    expect(result).toBeFalse();
+    expect(isConditionMet).not.toHaveBeenCalled();
+    expect(logConditionNotMetSpy).not.toHaveBeenCalled();
+    expect(logConditionErrorSpy).toHaveBeenCalledWith(
+      rule.conditions[0],
+      rule,
+      new Error(
+        'Rule component sequencing must be enabled on the property ' +
+          'for this condition to function properly.'
+      )
+    );
   });
 
-  it('calls logConditionNotMet when conditions are not met', function () {
+  it('handles a condition not being met', function () {
```

Comment:
This test case is the result of merging two test cases that were duplicating the same scenario but that had two different assertions.